### PR TITLE
Undo removal of latent heat transfer to compute the coil state in HexElementLatent 

### LINKED ISSUE #3065: Investigate WetCoilCounterFlow energy balance
**Describe the bug**

The commit df7415ebeb introduced some changes that break the energy balance of the model `Fluid.HeatExchangers.WetCoilCounterFlow` under the conditions simulated in `Fluid.HeatExchangers.Validation.WetCoilEffectivenessNTUCounterFlow`.
Basically the heat flow rate on the water side `*Q1_flow` remains equal to the sensible heat flow rate, leaving the latent part unbalanced, see attached plot.

WetCoilEffectivenessNTUCounterFlow

The main change was to remove the heat connection to the mass (heaConVapCoi), see details at https://github.com/lbl-srg/modelica-buildings/issues/3027#issuecomment-1138849935.

**Expected behavior**

`*Q1_flow` should be the opposite of `*Q2_flow`.

**Version**
 - Modelica Buildings Library: a4bf2006b2

### PR DESCRIPTION
**Pull request by AntoineGautier**
This closes #3065.

### CODE REVIEW

**Comment by dhblum** [comment]
@AntoineGautier Thanks again for the PR.  Looks good to me, I just made a couple more revisions to the documentation, merged the latest master, and resolved some conflicts.  I want to see how the unit tests fair in travis with the master update and will make any needed additional corrections once I see the results.

**Comment by dhblum** [comment]
@mwetter will review and then I will propagate changes also to maint_8.1.x and maint_9.0.x.

File: Buildings/Fluid/HeatExchangers/WetCoilCounterFlow.mo [hunk 1]
```diff
@@ -77,6 +77,13 @@ Buildings.Fluid.HeatExchangers.DryCoilCounterFlow</a> instead of this model.
 </html>", revisions="<html>
 <ul>
 <li>
+July 5, 2022, by Antoine Gautier:<br/>
+Restored the addition of heat to <code>mas.T</code> in
+<a href=\"Buildings.Fluid.HeatExchangers.BaseClasses.HexElementLatent\">
```
Comment (mwetter) [inline Buildings/Fluid/HeatExchangers/WetCoilCounterFlow.mo:82]: Such hyperlinks need to be `<a href=\"modelica://Buildings....`

**Review by mwetter — APPROVED** [review]

**Comment by mwetter** [comment]
@dhblum : The changes look good (apart from a wrong url format). I don't know why the CI test suddenly fails:[
https://github.com/lbl-srg/modelica-buildings/actions/runs/3365717460/jobs/5581467957
That is surprising.

**Comment by dhblum** [comment]
Thanks @mwetter.  Yea I'm not sure why they're failing.  Something with github actions repo versions.  But I can't quite figure out what it is.

**Comment by mwetter** [comment]
@dhblum : It is now fixed. I will go ahead and merge it.

**Pull request merged**